corehq/util/timer.py

```diff
@@ -175,3 +176,20 @@ def to_list(self, exclude_root=False):
 
 class TimerError(Exception):
     pass
+
+
+def time_method():
+    """Decorator to get timing information on a class method
+
+    The class must have a TimingContext instance as self._timing_context
+    """
+    def decorator(fn):
+        @wraps(fn)
+        def _inner(self, *args, **kwargs):
+            if self._timing_context:
+                with self._timing_context(fn.__name__):
+                    return fn(self, *args, **kwargs)
+            else:
+                return fn(self, *args, **kwargs)
+        return _inner
+    return decorator
```

Comment:
I didn't want to put `with self.timing_context('name'):` at the top of every method, and I thought this would be a decent way to abstract it.  The subsequent commit shows it in action.